tools/semgrep.yml

```diff
@@ -52,27 +50,11 @@ rules:
 
   - id: logging-format
     languages: [python]
-    pattern-either:
-      - pattern: logging.debug(... % ...)
-      - pattern: logging.debug(... .format(...))
-      - pattern: logger.debug(... % ...)
-      - pattern: logger.debug(... .format(...))
-      - pattern: logging.info(... % ...)
-      - pattern: logging.info(... .format(...))
-      - pattern: logger.info(... % ...)
-      - pattern: logger.info(... .format(...))
-      - pattern: logging.warning(... % ...)
-      - pattern: logging.warning(... .format(...))
-      - pattern: logger.warning(... % ...)
-      - pattern: logger.warning(... .format(...))
-      - pattern: logging.error(... % ...)
-      - pattern: logging.error(... .format(...))
-      - pattern: logger.error(... % ...)
-      - pattern: logger.error(... .format(...))
-      - pattern: logging.critical(... % ...)
-      - pattern: logging.critical(... .format(...))
-      - pattern: logger.critical(... % ...)
-      - pattern: logger.critical(... .format(...))
+    patterns:
+      - pattern-either:
+          - pattern: $X.$Y(... % ...)
+          - pattern: $X.$Y(... .format(...))
+      - pattern-where-python: "vars['$X'] in ['logging', 'logger'] and vars['$Y'] in ['debug', 'info', 'warning', 'error', 'critical']"
```

Comment:
So, I'd sorta expect this new pattern to be really slow performance-wise, because we're now going to check every `foo[.]bar` i the codebase.  Probably fixable by hardcoding the `$X` being logger/logging.

Reply:
Updated this with your suggestion. It surprisingly reduces runtime by 0.5s.